### PR TITLE
allow to set SNMPv3 context name per device

### PR DESCRIPTION
**Pull request by ohrensessel**
Context name is used as parameter to snmpwalk/get/...
Helpful for scenarios where SNMPv3 contexts are used for proxying

This interferes with #2820: Schema change and changes in functions/snmp.inc.php are partly taken from there. Did not see a nice possibility to cherry-pick, sorry.

Let me know what I can improve.

### CODE REVIEW

File: html/pages/addhost.inc.php [hunk 1]
```diff
@@ -46,6 +46,7 @@
                    'authalgo'   => mres($_POST['authalgo']),
                    'cryptopass' => mres($_POST['cryptopass']),
                    'cryptoalgo' => mres($_POST['cryptoalgo']),
+				   'context_name' => mres($_POST['context_name']),
```
Comment (SaaldjorMike) [inline html/pages/addhost.inc.php]: Looks like indentation is a mix of tabs and spaces here. Should be all spaces.

**Comment by laf** [comment]
I'm unsure if you mean that you've taken some code from #2820 or not with this? Either way we should be merging that PR in soon so this will need to wait until after that as it's been outstanding a while now so it's only fair to merge that PR and have you rebase this one.

**Comment by ohrensessel** [comment]
Yes, I have taken some code from #2820. (three lines within smtp.inc.php slightly altered by me, schema change adding context_name to devices table).

I will rebase my PR as soon as #2820 is merged.

Do you have any other comments regarding my PR? I'm thinking about making it more clear that Context Name is totally optional.

**Comment by laf** [comment]
We've now merged #2820. Can you rebase please.

**Comment by f0o** [comment]
Bump, please rebase and rename the Schema to `103`

**Comment by laf** [comment]
Schema 102, we have one waiting on 103 already.

**Comment by laf** [comment]
oh lol. just seen your other comment. 103 it is then

**Comment by laf** [comment]
This needs to be 104.sql. Can you rebase please.

**Comment by ohrensessel** [comment]
Yes. Sorry a little low on time
Am 23.02.2016 23:45 schrieb "Neil Lathwood" <notifications@github.com>:

**Comment by laf** [comment]
Needs to be 105.sql now.

**Comment by f0o** [comment]
Needs to be 109.sql now

**Comment by laf** [comment]
bump, please rebase.

File: includes/snmp.inc.php [hunk 2]
```diff
@@ -752,12 +752,12 @@ function snmp_gen_auth(&$device) {
...
-		
-		//add context if exist context
-		if(key_exists('context_name', $device)){
-			$cmd = " -v3 -n '".$device['context_name']."' -l '".$device['authlevel']."'";
-		}
-		
+
+        //add context if exist context
+        if(key_exists('context_name', $device) && !empty($device['context_name'])){
+            $cmd = " -v3 -n '".$device['context_name']."' -l '".$device['authlevel']."'";
```
Comment (laf) [inline includes/snmp.inc.php]: We already specify -v3 -n on line 754. What happens when it's repeated?

Comment (laf) [inline includes/snmp.inc.php]: Looks we probably should do an else and add the defaults below

**Comment by laf** [comment]
Thanks for updating, just put a comment in with regards to a bit of code.

**Comment by laf** [comment]
Also, you will need to sign the contributors agreement as per http://docs.librenms.org/General/Contributing/

**Comment by laf** [comment]
bump

**Comment by laf** [comment]
bump

**Comment by laf** [comment]
Closing, feel free to reopen this when you've rebased.

**Pull request closed without merging**